gui/src/app/view/transactions.rs

```diff
@@ -157,6 +157,70 @@ fn tx_list_view(i: usize, tx: &HistoryTransaction) -> Element<'_, Message> {
     .into()
 }
 
+pub fn create_rbf_modal<'a>(
+    is_cancel: bool,
+    feerate: &form::Value<String>,
+    replacement_txid: Option<Txid>,
+    warning: Option<&'a Error>,
+) -> Element<'a, Message> {
+    let mut confirm_button = button::primary(None, "Confirm").width(Length::Fixed(200.0));
+    if feerate.valid || is_cancel {
+        confirm_button =
+            confirm_button.on_press(Message::CreateRbf(super::CreateRbfMessage::Confirm));
+    }
+    let help_text = if is_cancel {
+        "Replace the transaction with one paying a higher feerate that sends the coins back to us. There is no guarantee the original transaction won't get mined first. New inputs may be used for the replacement transaction."
+    } else {
+        "Replace the transaction with one paying a higher feerate to incentivize faster confirmation. New inputs may be used for the replacement transaction."
+    };
+    card::simple(
+        Column::new()
+            .spacing(10)
+            .push(Container::new(h4_bold("Transaction replacement")).width(Length::Fill))
+            .push(Row::new().push(text(help_text)))
+            .push_maybe(if !is_cancel {
+                Some(
+                    Row::new()
+                        .push(Container::new(p1_bold("Feerate")).padding(10))
+                        .spacing(10)
+                        .push(
+                            form::Form::new_trimmed("", feerate, move |msg| {
```

Comment:
It might be better not to trim the feerate value as it's difficult to edit a single-digit value. It's not possible to delete the existing value and enter a new one and we have to instead add a second digit and delete the first.